test/Index/conformances.swift

```diff
@@ -79,6 +79,12 @@ class SubMultiConf: BaseMultiConf,P2,P1,P3 { // CHECK: [[@LINE]]:7 | class/Swift
   // CHECK-NOT: [[@LINE-13]]:7 | instance-method
 }
 
+class CompositionType: BaseMultiConf & P1 { // CHECK: [[@LINE]]:7 | class/Swift | CompositionType | [[CompositionType_USR:.*]] | Def
```

Comment:
Could you also add a test for a typealias? We'll still be missing those relations IIUC, but it seems reasonable to also handle. The relations should be implicit for that case though.

`reportRelatedTypeRef` currently adds a relation for the underlying type of a typealias if it's a nominal type, but not otherwise. We'd need an extra `isImplicit` on `reportRelatedTypeRef`, but seems like we could handle `CompositionTypeRepr` in there rather than on `reportInheritedTypeRefs` - then the `TypeAliasDecl` would just call `reportRelatedTypeRef` again with the underlying type and `isImplicit` set.

Reply:
thanks for the pointer, new version here has support for this and tests for it. the gist is what you said with 2 other required changes:

1. I had to pass through the location where the annotation should be, since that would otherwise use the declaration location
2. I handled similar logic in `visitDeclReference` to get the same behavior there, and for recursive typealiases